components/skus/renderer/skus_utils.cc

```diff
@@ -23,8 +23,14 @@ bool IsSafeOrigin(const blink::WebSecurityOrigin& origin) {
        {blink::WebSecurityOrigin::Create(
            GURL("https://account.brave.software"))}}};
 
+  const GURL origin_url(origin.ToString().Utf8());
+  const GURL stripped_port_url =
+      GURL(origin_url.scheme() + "://" + origin_url.host() + origin_url.path());
+  blink::WebSecurityOrigin stripped_port_origin =
+      blink::WebSecurityOrigin::Create(stripped_port_url);
+
   for (const blink::WebSecurityOrigin& safe_origin : *safe_origins) {
-    if (safe_origin.IsSameOriginWith(origin)) {
+    if (safe_origin.IsSameOriginWith(stripped_port_origin)) {
       return true;
     }
```

Comment:
Code for test-only purposes should not affect production behavior. To handle random port of embedded server use something like this in SetUpCommandLine:

```    
command_line->AppendSwitchASCII(
        network::switches::kHostResolverRules,
        base::StringPrintf("MAP account.brave.software:443 127.0.0.1:%d",
                           https_server_.port()));
```
Or search for `kHostResolverRules` for more examples.

Reply:
I'm not sure this will work in this case. If I understand right, the port of the`origin` parameter in IsSafeOrigin is still the randomly assigned value, even if we route traffic to account.brave.software:443 to the randomly generated port. I tried starting the https_server_ at 443, but I get a permissions denied error.